randomstate/interface/dSFMT/dSFMT-shim.h

```diff
@@ -67,8 +67,11 @@ static NPY_INLINE double random_double(aug_state *state) {
 }
 
 static NPY_INLINE uint64_t random_raw_values(aug_state *state) {
+  uint64_t out;
   double d = random_double_from_buffer(state);
-  return *((uint64_t *)&d);
+
+  memcpy(&out, &d, sizeof(d));
```

Comment:
Wouldn't this add significant overhead to this method?

I'm just curious as I would naturally have avoided what I would assume would just be unnecessary overhead, but my C is pretty terrible and likely full of bad habits.

Reply:
Good question -- in theory the compiler could generate the same code since the sizeof(d) is known.  I'll have to write a benchmark to check.

Reply:
I just tested this and there is no performance difference -- if anything there was a tiny advantage tot the new method, although this was probably sampling error.

Reply:
Good to know. One of my go-to techniques to improve performance is to try and eliminate redundant allocations but I guess in this case the compiler is smart enough to do that itself.